Bumped Gradle, Loom, dependencies versions

lospejos opened the pull request.

lospejos committed:
Bumped Gradle, Loom, dependencies versions

Noaaan commented:
This PR seems to do a little bit more than just bump versions, like pre-maturely bumping the main mod version and including static analysis. Care to describe why?

lospejos commented:
I updated the minor version of mod to reflect that this build has slight changes comparing to previous version.

Regarding including static analysis I'm afraid didn't get you properly.
If you meant adding a `findbug` dependency, I made it because in modern Java versions some annotations were removed (particularly NotNull). I've added `findbug` dependency to support removed annotations.

Noaaan commented on build.gradle:
```diff
@@ -1,7 +1,7 @@
+    id 'fabric-loom' version '1.3-SNAPSHOT'
     id 'maven-publish'
-    id 'io.github.juuxel.loom-quiltflower' version '1.7.2'
+    id 'io.github.juuxel.loom-quiltflower' version '1.10.0'
```
As of 1.11.0 this has been rebranded to vineflower, use that instead. 
https://github.com/Juuxel/loom-vineflower

Noaaan commented on build.gradle:
```diff
@@ -21,6 +21,11 @@ repositories {
     maven { url = "https://api.modrinth.com/maven/" }
     maven { url = "https://maven.nucleoid.xyz/" }
 
+    maven {
+        name = "TerraformersMC"
+        url = "https://maven.terraformersmc.com/"
+    }
+
```
Remove this. It is a duplicate of line 15

Noaaan commented on build.gradle:
```diff
@@ -79,6 +84,8 @@ dependencies {
     modImplementation include("dev.onyxstudios.cardinal-components-api:cardinal-components-base:${project.cca_version}")
     modImplementation include("dev.onyxstudios.cardinal-components-api:cardinal-components-entity:${project.cca_version}")
 
+    implementation 'com.google.code.findbugs:jsr305:3.0.2'
+
```
I will admit that the inclusion of annotation sources is not 100% consistent in this project. I would however much prefer to not include an extra dependency for the sole reason of bringing in annotations. The only missing annotation in this particular case is `@Nullable`, which can be sourced from the included JetBrains annotations dep (which in this project is used heavily already).

Noaaan commented on gradle.properties:
```diff
@@ -43,7 +41,11 @@ aea_versionite=1.4.0+1.20.0
-cpa_version = 1.0.0
\ No newline at end of file
+cpa_version = 1.0.0
+
```
Nitpick: remove this newline to be consistent with the rest

Noaaan commented on gradle.properties:
```diff
@@ -8,33 +8,31 @@ yarn_mappings=1.20.1+build.9
 
 # Mod Properties
-mod_version = 0.18.2+1.20
+mod_version = 0.18.3+1.20.1
```
Please do not bump the mod version for this project. 
Currently I support Minecraft 1.20 (1.20.0) and 1.20.1 if applicable. I do not know whether you have tested on said version, but if it works then the appended data on the semver is wrong. If, and only if, this version does not work on 1.20, then this would also require a change in the `fabric.mod.json` instead. 
Either way, I have a large list of fixes and plans, which might cause this to become a major patch instead of minor. I therefore prefer to wait with versions bumps until I have planned out a release.

Noaaan reviewed (changes requested):
See attached review comments

lospejos committed:
Bumped Gradle, Loom, dependencies versions, also fixed as suggested by repo owner

lospejos commented:
> See attached review comments

Thanks for your review, I appreciate it. I suppose that I've fixed all by your comments and I made a new commit.
